content/hardware/04.pro/boards/portenta-x8/tutorials/datalogging-iot/content.md

```diff
@@ -0,0 +1,416 @@
+
+```arduino
+#include <ArduinoMqttClient.h>
+#include <WiFiNINA.h>
+#include "arduino_secrets.h"
+
+// Wi-Fi information
+char ssid[] = SECRET_SSID;        // Your network SSID, stored in the arduino_secrets.h file
+char pass[] = SECRET_PASS;        // Your network password, stored in the arduino_secrets.h file
+WiFiClient wifiClient;
+MqttClient mqttClient(wifiClient);
+
+// MQTT broker information
+
+const char broker[] = "your-portenta-ip";
+int        port     = 1883;
+const char topic[]  = "test";
+
+// Interval for sending messages (in milliseconds) to the MQTT broker
+const long interval = 5000;
+unsigned long previousMillis = 0;
+
+// Data to send to the MQTT broker
+int count = 0;
+
+void setup() {
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // Wait for serial port to connect, needed for native USB port only
+  }
+
+  // Attempt to connect to the defined Wi-Fi network
+  Serial.print("- Attempting to connect to WPA SSID: ");
+  Serial.println(ssid);
+
+  while (WiFi.begin(ssid, pass) != WL_CONNECTED) {
+    // Connection attempt failed, retry again
+    Serial.print(".");
+    delay(5000);
+  }
+
+  Serial.println("- You're connected to the network!");
+  Serial.println();
+  
+  // Attempt to connect to the defined Wi-Fi network
+  Serial.print("- Attempting to connect to the MQTT broker: ");
+  Serial.println(broker);
+
+  // Connection attempt to the MQTT broker failed
+  if (!mqttClient.connect(broker, port)) {
+    Serial.println("- MQTT connection failed!");
+    Serial.print("- Error code: ");
+    Serial.println(mqttClient.connectError());
+
+    while (1);
+  }
+
+  Serial.println("- You're connected to the MQTT broker!");
+  Serial.println();
+}
+
+void loop() {
+  // Keep the board connected to the MQTT broker
+  mqttClient.poll();
+
+  unsigned long currentMillis = millis();
+
+  if (currentMillis - previousMillis >= interval) {
+    // Send the gathered data to an specific topic of the MQTT broker
+    previousMillis = currentMillis;
+    Serial.print("- Sending message to topic: ");
+    Serial.println(topic);
+    Serial.print("- ");
+    Serial.println(count);
+    mqttClient.beginMessage(topic);
+    mqttClient.print(count);
+    mqttClient.endMessage();
+    Serial.println();
+
+    // Update data value    
+    count++;
+  }
+}
+```
+
+The sketch shown above connects the MKR WiFi 1010 to the local MQTT broker of the X8 and periodically sends the value of a counter to the topic `test`. 
+
+***Please read [this tutorial](https://docs.arduino.cc/tutorials/mkr-wifi-1010/mqtt-device-to-device#programming-the-publisher) for more in-depth information about MQTT and the MKR WiFi 1010 board.***
+
+If everything is ok, you should see the following in the Serial monitor of the Arduino IDE:
+
+**ADD IMAGE HERE**
+
+Now let's check out out Grafana dashboard, we should see data from the MKR WiFi 1010 board:
+
+**ADD IMAGE HERE**
+
+## Conclusion
+
+### Next Steps
```

Comment:
Please ping me out once the Last section its finished! 👀

Reply:
Ping ping ping 👀